### ISSUE TITLE
Shared Alarm flow not working (stuck at Google Sign In) 

### ISSUE THREAD

**Issue by vici14** [CCExtractor/ultimate_alarm_clock#117]
### Description

I've played the app and it seems Google Sign-In flow for Shared Alarm not working

Reproduce:

Tap Enable Shared Alarm -> Show popup Disabled! -> Go to Settings ->Sign-In with Google -> Logged In GG account -> Show popup Success -> back to home -> Tap Enable Shared Alarm -> Show popup Disabled!

Result : 
Show popup Disabled after Logged in Google account

Expected Result:
Tap Enable Shared Alarm after Logged in Google does not show popup disabled and  start flow Shared Alarm such as :
-Share DynamicLinks / DeepLink to users invited (feat)
-Using CloudFunction(Firebase) to trigger a notification that they have received an invite for SharedAlarm and request them to Log in Google account (feat)
-Users who received an invite can accept or cancel 
-Users who received an invite should have a List Shared Alarm from friends, each update on this Alarm should be notify back to whom created(feat)

The expected result I listed actually should break into Feature requests I guess. I'm new commer in this project so I haven't had access to resources such as Firebase and GCP (if they're available)

Please discuss further information to make the implementation planning well.

**Comment by MarkisDev** [comment]
Hey, could you attach a video demonstrating the bug?

As for the feature, I think it's a great idea. Using CF should make this doable for sure, but we can start off using a share intent to share the alarm id. Please go ahead and make a new issue for this feat.

**Comment by vici14** [comment]
https://github.com/CCExtractor/ultimate_alarm_clock/assets/33067303/4a917fd1-df8d-42cf-a660-d5a050ce2ae7

Here's a video demonstrating bug

**Comment by MarkisDev** [comment]
Alright! I'll assign this issue to you - feel free to ask anything if you get stuck :)

**Comment by vici14** [comment]
Thanks, firstly I need to be invited to this Firebase  and Google Cloud Platform project, I don't know if you created it yet, here's my email:cuongchautech14@gmail.com

**Comment by vici14** [comment]
@MarkisDev Hi, i haven't been granted in Firebase project, can you invite me when you're free

**Comment by MarkisDev** [comment]
@vici14 I've added you to the project. If you're going to add/remove something, please discuss it with me beforehand :)

Would be better to keep this feat as a separate issue!

**Comment by MarkisDev** [comment]
I'm so sorry for the oversight! It seems I added you to a different project! 

I've added you to the right one now (and yes I double-checked this time!)

**Comment by vici14** [comment]
Hi @MarkisDev, I figured out that the issue will happen when another new developers working on their machine, build app with debug, release mode without adding the SHA1 debugkey on Firebase will cause the issue (com.google.android.gms.common.api.ApiException: 10:).I've solved login issue by added SHA1 key to Firebase, and have tested on another machine without added SHA1 key.
I suggest we separate into 2 Firebase project with corresponding app's flavor, 1 is for development, 1 is for production, newcomers want to working on Firebase,Google apis functions should add their SHA key to the development project.Otherwise, production project only for Google Play Store key. This will prevent the misleading and conflict keys on the Firebase project.
This issue should be closed and i will create feature request for prepare the dev and prod environment?
